services/core-web/src/components/dashboard/minesHomePage/MineList.js

```diff
@@ -1,191 +1,188 @@
     dataIndex: "tenure",
     width: 150,
-    render: (text, record) => (
+    render: (text) => (
       <div title="Tenure">
-        {text &&
-          text.map((tenure) => (
-            <span className="mine_tenure" key={tenure.mine_type_guid}>
-              {record.tenureHash[tenure.mine_tenure_type_code]}
-            </span>
-          ))}
-        {!text && <div>{record.emptyField}</div>}
+        {(text && text.length > 0 && text.join(", ")) || Strings.EMPTY_FIELD}
       </div>
     ),
   },
   {
     title: "Commodity",
+    key: "commodity",
     dataIndex: "commodity",
     width: 150,
-    render: (text, record) => (
+    render: (text) => (
       <div title="Commodity">
-        {(text &&
-          text
-            .map(({ mine_type_detail }) =>
-              mine_type_detail
-                .map(
-                  ({ mine_commodity_code }) =>
-                    mine_commodity_code && record.commodityHash[mine_commodity_code]
-                )
-                .filter(Boolean)
-                .join(", ")
-            )
-            .join(", ")) ||
-          record.emptyField}
+        {(text && text.length > 0 && text.join(", ")) || Strings.EMPTY_FIELD}
       </div>
     ),
   },
   {
     title: "TSF",
+    key: "tsf",
     dataIndex: "tsf",
     width: 150,
     render: (text) => <div title="TSF">{text}</div>,
   },
 ];
 
-const transformRowData = (mines, mineIds, mineRegionHash, mineTenureHash, mineCommodityHash) =>
-  mineIds
-    .filter((id) => mines[id])
-    .map((id) => ({
-      key: id,
-      emptyField: Strings.EMPTY_FIELD,
-      mineName: mines[id].mine_name ? mines[id].mine_name : Strings.EMPTY_FIELD,
-      mineNo: mines[id].mine_no ? mines[id].mine_no : Strings.EMPTY_FIELD,
-      operationalStatus:
-        mines[id].mine_status &&
-        mines[id].mine_status[0] &&
-        mines[id].mine_status[0].status_labels &&
-        mines[id].mine_status[0].status_labels[0]
-          ? mines[id].mine_status[0].status_labels[0]
-          : Strings.EMPTY_FIELD,
-      permit:
-        mines[id].mine_permit_numbers && mines[id].mine_permit_numbers[0]
-          ? mines[id].mine_permit_numbers
-          : null,
-      region: mines[id].mine_region ? mineRegionHash[mines[id].mine_region] : Strings.EMPTY_FIELD,
-      commodity: mines[id].mine_type && mines[id].mine_type[0] ? mines[id].mine_type : null,
-      commodityHash: mineCommodityHash,
-      tenure: mines[id].mine_type && mines[id].mine_type[0] ? mines[id].mine_type : null,
-      tenureHash: mineTenureHash,
-      tsf: mines[id].mine_tailings_storage_facilities
-        ? mines[id].mine_tailings_storage_facilities.length
+const transformRowData = (mines, mineRegionHash, mineTenureHash, mineCommodityHash) =>
+  Object.values(mines).map((mine) => ({
+    key: mine.mine_guid,
+    mine_name: mine.mine_name || Strings.EMPTY_FIELD,
+    mine_no: mine.mine_no || Strings.EMPTY_FIELD,
+    mine_operation_status_code:
+      mine.mine_status &&
```

Comment:
Possibly able to use lodash safe navigation get as:
`_.get('mine.mine_status[0].status_labels[0]')` 

Just learned about this https://masteringjs.io/tutorials/lodash/get